Bottom: df1f7ba9c19c6cabb55006077a2d7b0066f8dc79
Top:    0069ad432db6f2e9da7c6cd580a79b4e212155b3
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-12-27 19:33:39 +0000

Fix to ParseGroupString()


---

diff --git a/src/misc.c b/src/misc.c
index 10796fc..63d7ca4 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1187,7 +1187,14 @@ ParseGroupString (char *group_string, LayerGroupType *LayerGroup, int *LayerN)
           /* ignore white spaces and check for separator */
           while (*s && isspace ((int) *s))
             s++;
-          if (!*s || *s == ':')
+
+          /* Backup so the outer loop terminates on the NULL termination */
+          if (*s == '\0') {
+            s--;
+            break;
+          }
+
+          if (*s == ':')
             break;
         }
       LayerGroup->Number[group] = member;
